luminoth/tools/dataset/readers/object_detection/openimages.py

```diff
@@ -0,0 +1,226 @@
+import csv
+import os
+import six
+import tensorflow as tf
+import threading
+
+from six.moves import queue
+from PIL import Image
+
+from luminoth.tools.dataset.readers import InvalidDataDirectory
+from luminoth.tools.dataset.readers.object_detection import (
+    ObjectDetectionReader
+)
+from luminoth.utils.dataset import read_image
+
+CLASSES_TRAINABLE = 'classes-bbox-trainable.txt'
+CLASSES_DESC = 'class-descriptions.csv'
+ANNOTATIONS_FILENAME = 'annotations-human-bbox.csv'
+IMAGES_LOCATION = 'gs://open-images-dataset'
+
+
+class OpenImagesReader(ObjectDetectionReader):
+    """OpenImagesReader supports reading the images directly from the original
+    data source which is hosted in Google Cloud Storage.
+
+    Before using it you have to request and configure access following the
+    instructions here: https://github.com/cvdfoundation/open-images-dataset
+    """
+    def __init__(self, data_dir, split, download_threads=25, **kwargs):
+        super(OpenImagesReader, self).__init__()
+        self._data_dir = data_dir
+        self._split = split
+        self._download_threads = download_threads
+
+        self._classes = None
+        self._total = None
+        self._image_ids = None
+
+        self.yielded_records = 0
+        self.errors = 0
+
+    @property
+    def classes(self):
+        if self._classes is None:
+            trainable_labels_file = os.path.join(
+                self._data_dir, CLASSES_TRAINABLE)
+            trainable_labels = set()
+            try:
+                with tf.gfile.Open(trainable_labels_file) as tl:
+                    for label in tl:
+                        trainable_labels.add(label.strip())
+            except tf.errors.NotFoundError:
+                raise InvalidDataDirectory(
+                    'Missing label file "{}" from data_dir'.format(
+                        CLASSES_TRAINABLE))
+
+            labels_descriptions_file = os.path.join(
+                self._data_dir, CLASSES_DESC)
+            desc_by_label = {}
+            try:
+                with tf.gfile.Open(labels_descriptions_file) as ld:
+                    reader = csv.reader(ld)
+                    for line in reader:
+                        if line[0] in trainable_labels:
+                            desc_by_label[line[0]] = line[1]
+            except tf.errors.NotFoundError:
+                raise InvalidDataDirectory(
+                    'Missing label description file "{}" from data_dir'.format(
+                        CLASSES_DESC))
+
+            self._classes = sorted(trainable_labels)
+            self._descriptions = [
+                desc for _, desc in
+                sorted(desc_by_label.items(), key=lambda x: x[0])
```

Comment:
Prefer to use `itemgetter(0)` instead of lambda. [Docs](https://docs.python.org/2/library/operator.html#operator.itemgetter).